src/Raven.Client/Http/RequestExecutor.cs

```diff
@@ -1337,8 +1352,8 @@ public bool Translate(object sender, X509Certificate cert, X509Chain chain, SslP
         {
             add
             {
-                if (PlatformDetails.RunningOnMacOsx)
-                    throw new PlatformNotSupportedException("On Mac OSX, is it not possible to register to RemoteCertificateValidationCallback because of https://github.com/dotnet/corefx/issues/9728");
+                if (ServerCertificateCustomValidationCallbackRegistrationException != null)
+                    throw ServerCertificateCustomValidationCallbackRegistrationException;
```

Comment:
Please wrap it. I'm not sure whatever this can cause issue, but throwing the same exception concurrently sounds like potential race condition

Reply:
Wrapped :)